### DISCUSSION TITLE
YOLOv10 on artpec8

### DISCUSSION THREAD

**Discussion by p-robak97** [AxisCommunications/acap-computer-vision-sdk-examples#193] [Q&A]
I have succesfully exported yolov10 model to full int8 quant tflite model using ai-edge-torch. I tired to do the inference using modified pose-estimator-with-flask but i get the following error from acap-runtime:
```
inference-server-1     | ERROR in Inference: Failed to load model model.tflite (Could not load model: Could not allocate tensors)
inference-server-1     | INFO in acapruntime: Exit /opt/app/acap_runtime/acapruntime
```

then in `journalctl` i found that:
```
Sep 13 12:51:11 axis-b8a44fa15b7c larod[20779]: /usr/src/debug/tensorflow-lite/2.10.1/tensorflow/lite/kernels/concatenation.cc:158 t->dims->data[d] != t0->dims->data[d] (16 != 20)
Sep 13 12:51:11 axis-b8a44fa15b7c larod[20779]: Node number 117 (CONCATENATION) failed to prepare.
Sep 13 12:51:11 axis-b8a44fa15b7c systemd[1]: mnt-flash-var-lib-larod\x2dswap-swap.swap: Deactivated successfully.
Sep 13 12:51:11 axis-b8a44fa15b7c larod[20779]: Session 0: Could not load model: Could not allocate tensors
Sep 13 12:51:11 axis-b8a44fa15b7c larod[20779]: Session 0 killed since client's (:1.604) connection has been lost
```

The problem seems to be in CONCATENATION. I checked locally using tf lite Interpreter and the model works fine with tensorflow>=2.14 but i get the same error with older versions of tensorflow. The error from larod suggests there is tensorflow-lite 2.10.1 version. I tried to build my own acap-runtime image with tensorflow 2.14 but this does not resolve my problem as i'm getting the same error.

Is it possible to upgrade tensorflow lite to >=2.14?

Also i don't understand why the error suggests that tensorflow-lite version is 2.10.1, when in all docker containers there is 2.9.

Environment

    Axis device model: P3265-LVE
    Axis device firmware version: 11.11.73
    SDK VERSION: 1.15
    docker-acap: 3.0.1-rc.1
    acap-runtime: 1.3.1 and with modified tensorflow and tensorflow serving version to 2.14

**Comment by Corallo** [comment]
@p-robak97 Tensorflow of the inference server is only used to build the protofiles to allow communication between the application container and the inference server. You can find out the Tensorflow version that is actually in the firmware by looking at the SBOM file that comes when you download an Axis OS version. More details [here](https://developer.axis.com/computer-vision/computer-vision-on-device/supported-frameworks)
